### ISSUE TITLE
RegReplace stops working if "selection_only":true

### ISSUE THREAD

**Issue by pierodancona** [facelessuser/RegReplace#56]
I installed RegReplace via Package Control on Mac OSX
It works perfectly (thank you!), but after creating the file
reg_replace.sublime-settings in my User directory, 
which contains only
      {      "selection_only": true      }
and some text is selected, RegReplace becomes
ineffective. When no text is selected, it works
as usual on the whole file.

If I replace the content of the settings file with
      {          }
or with
      {      "selection_only": false      }
then the plugin works again as expected.

---------------------------------
PS This is probably unrelated, but when starting up ST3 the
following error appears:

.....
reloading plugin RegReplace.spellcheck
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 109, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 915, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "spellcheck in /Users/piero/Library/Application Support/Sublime Text 3/Installed Packages/RegReplace.sublime-package", line 6, in <module>
ImportError: No module named 'yaml'
reloading plugin RegReplace.support
....

**Comment by facelessuser** [comment]
1. Please do follow the issue template and provide version number etc.  Provide example regex and some example source to apply the regex.  Describe how you run the command etc.  Essentially what the issue template asks for: https://github.com/facelessuser/RegReplace/blob/master/.github/ISSUE_TEMPLATE.md.  It is difficult for me to guess what people are doing that may be contributing to their issues without detailed issue reports.

2. >PS I do not think this is related, but when starting up ST3 the
following error appears:

    Yeah, I know what this is.  I need to move the spellcheck module under tests so Sublime doesn't try to load it as a plugin module.

**Comment by pierodancona** [comment]
Ok. Here is the result of the Package Info command:
- ST ver.: 3125
- Platform: osx
- Arch: x64
- Plugin ver.: 3.2.1
- Install via PC: True
- mdpopups ver.: 1.13.1
- backrefs ver.: 1.0.post1
- regex ver.: 2.4.112
- markdown ver.: 2.6.6
- pygments ver.: 2.1a0
- jinja2 ver.: 2.8

Complete example:
1) file reg_replace_rules.sublime-settings contains only the line
{"replacements": {"aacute_to_latex": {"scope": "text", "find": "á", "replace": "\\'{a}"}}}

2) file reg_replace.sublime-settings contains only the line
{"selection_only": true }

3) file Default.sublime-commands contains the command definition
{
    "caption": "Reg Replace: Lettere accentate to LaTeX",
    "command": "reg_replace",
    "args": {"replacements": ["aacute_to_latex"]},
},

4) plain text file contains only one line (four accented a's)
á   á   á   á   

5) I select the first two accented a's and run the command via
Command Palette: nothing happens. However, if I select the
whole line or if nothing is selected, the command works fine
on all letters

6) I change the setting to {"selection_only": false }. Now the
command works as expected

**Comment by facelessuser** [comment]
Okay.  When doing a scope search (which is what you are doing when you specify `scope`), RegReplace looks for all the scopes, then applies regex to those scopes.  RegReplace currently, when `selection_only` is `True` and there are selections, requires the selection to contain the whole scope.  In the future this may change, but currently that is how it works.  That is why when you expand the selection to include the whole scope, it suddenly works.  Regex is optional with the scope searches.  The scope is the actual search result, but you can apply regex to it.

I'm not sure why you need to specify `scope` in your example.  Unless you have a reason to specify the scope, I would not specify the scope.  You could use `scope_filter` instead which works different.  When specifying `scope_filter`, the regex is found first, then the scope is checked on that match to see if it is a qualified match.  In this case the regex match must be in the selection, not the entire scope.

So in short, with scope searches, the scope is the target (then we apply regex to it -- if specified).  In normal searches, the search result is the target, and the scope is applied to validate the found result.

Hopefully that makes sense.

**Comment by pierodancona** [comment]
Oh I see. Indeed it works without scope.
Thank you!

**Comment by facelessuser** [comment]
I will go ahead and close this. What to do with scope searches is a separate issue that I need to think about.  It might not change, but it may in the future.
